arcade/sprite_list/sprite_list.py

```diff
@@ -249,10 +254,18 @@ def __iter__(self) -> Iterator[SpriteType]:
         """Return an iterable object of sprites."""
         return iter(self.sprite_list)
 
+    @overload
+    def __getitem__(self, i: int) -> SpriteType:
+        ...
+
+    @overload
+    def __getitem__(self, i: slice) -> List[SpriteType]:
+        ...
+
```

Comment:
These have no impact on performance?

Reply:
TL;DR Reverted for now

I was under the impression `SpriteList` indexing wasn't intended to be a primary organization method. The performance hit looks to be 7%.

[cProfile gist](https://gist.github.com/pushfoo/3bfa7aea2b2c916b2b0511b3ea3b6276)

### With `@overload`
```
20000001 function calls in 1.848 seconds

   Ordered by: standard name

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
 10000000    1.173    0.000    1.173    0.000 sprite_list.py:265(__getitem__)
 10000000    0.675    0.000    0.675    0.000 {built-in method builtins.next}
        1    0.000    0.000    0.000    0.000 {method 'disable' of '_lsprof.Profiler' objects}
```

### Without `@overload` (development)
```
20000001 function calls in 1.730 seconds

   Ordered by: standard name

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
 10000000    1.078    0.000    1.078    0.000 sprite_list.py:252(__getitem__)
 10000000    0.653    0.000    0.653    0.000 {built-in method builtins.next}
        1    0.000    0.000    0.000    0.000 {method 'disable' of '_lsprof.Profiler' objects}
```